agent/config/config_test.go

```diff
@@ -444,6 +444,179 @@ func TestGet(t *testing.T) {
 		assert.Equal(t, filepath.Join(wd, name), configFilepath)
 		assert.Equal(t, ConfigFileDoesNotExistError(filepath.Join(wd, name)), err)
 	})
+
+	t.Run("SameGRPCLocationCustomSocket", func(t *testing.T) {
+		actual, configFilepath, err := get([]string{
+			"--id=agent-id",
+			"--listen-socket=/tmp/pmm-agent.sock",
+			"--server-address=127.0.0.1",
+		}, logrus.WithField("test", t.Name()))
+		require.NoError(t, err)
+
+		expected := &Config{
+			ID:               "agent-id",
+			ListenSocket:     "/tmp/pmm-agent.sock",
+			ListenSocketGRPC: "/tmp/pmm-agent-grpc.sock",
```

Comment:
do we really need two separate files?

Reply:
I may be wrong but I could not find a way for the grpc gateway to talk directly to grpc server without exposing a socket or a port. For this reason we have two socket files.